eng/Versions.props

```diff
@@ -60,17 +60,17 @@
     -->
     <!-- Microsoft.CodeAnalysis versions for different purposes. -->
     <!-- Surface area that various projects compile against. These should have source-build reference packages -->
-    <MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>3.7.0</MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>
+    <MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>3.11.0</MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>
     <MicrosoftCodeAnalysisVersionForPublicApiAnalyzers>1.2.1</MicrosoftCodeAnalysisVersionForPublicApiAnalyzers>
-    <MicrosoftCodeAnalysisVersionForBannedApiAnalyzers>2.9.0</MicrosoftCodeAnalysisVersionForBannedApiAnalyzers>
-    <MicrosoftCodeAnalysisVersionForBannedApiAnalyzersTests>3.3.1</MicrosoftCodeAnalysisVersionForBannedApiAnalyzersTests>
-    <MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzers>2.9.0</MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzers>
-    <MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzersTests>3.3.1</MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzersTests>
+    <MicrosoftCodeAnalysisVersionForBannedApiAnalyzers>3.11.0</MicrosoftCodeAnalysisVersionForBannedApiAnalyzers>
+    <MicrosoftCodeAnalysisVersionForBannedApiAnalyzersTests>3.11.0</MicrosoftCodeAnalysisVersionForBannedApiAnalyzersTests>
+    <MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzers>3.11.0</MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzers>
+    <MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzersTests>3.11.0</MicrosoftCodeAnalysisVersionForPerfSensitiveAnalyzersTests>
     <MicrosoftCodeAnalysisVersionForResxSourceGenerators>4.0.1</MicrosoftCodeAnalysisVersionForResxSourceGenerators>
-    <MicrosoftCodeAnalysisVersionForNetAnalyzers>3.3.1</MicrosoftCodeAnalysisVersionForNetAnalyzers>
-    <MicrosoftCodeAnalysisVersionForTextAnalyzers>3.3.1</MicrosoftCodeAnalysisVersionForTextAnalyzers>
-    <MicrosoftCodeAnalysisVersionForCodeAnalysisAnalyzers>3.3.1</MicrosoftCodeAnalysisVersionForCodeAnalysisAnalyzers>
-    <MicrosoftCodeAnalysisVersionForToolsAndUtilities>3.3.1</MicrosoftCodeAnalysisVersionForToolsAndUtilities>
+    <MicrosoftCodeAnalysisVersionForNetAnalyzers>3.11.0</MicrosoftCodeAnalysisVersionForNetAnalyzers>
+    <MicrosoftCodeAnalysisVersionForTextAnalyzers>3.11.0</MicrosoftCodeAnalysisVersionForTextAnalyzers>
+    <MicrosoftCodeAnalysisVersionForCodeAnalysisAnalyzers>3.11.0</MicrosoftCodeAnalysisVersionForCodeAnalysisAnalyzers>
+    <MicrosoftCodeAnalysisVersionForToolsAndUtilities>3.11.0</MicrosoftCodeAnalysisVersionForToolsAndUtilities>
```

Comment:
@mmitche @MichaelSimons This PR moves the referenced `Microsoft.CodeAnalysis` versions for the packages to `3.11.0` (shipped with VS2019 16.11). However, this fails the pre-build detection step for source build that you added with #6610. I see that https://github.com/dotnet/source-build-reference-packages/pull/692 added the SB reference packages for the prior versions of the dependency packages, and are being mapped from here: https://github.com/dotnet/roslyn-analyzers/blob/b4dc141b0568be5aa135965d30c49c740b706be1/eng/Version.Details.xml#L8-L12

I presume this PR would require an update to the SB packages to add the `3.11.0` version of the packages before we can pass the pre-build detection leg. Is that correct? If so, who can help me out to perform this update in https://github.com/dotnet/source-build-reference-packages? If there are documentation pointers on how to go about this, please share them so I can try to do this myself. Thanks!

Reply:
Ah, I see the steps at https://github.com/dotnet/source-build-reference-packages/tree/main#adding-new-packages. I'll give it a shot.

Reply:
I tried executing those steps locally, but I get a build failure at the first step itself:

> generate.cmd -package microsoft.codeanalysis,3.11.0

**Error:**
```
  PackageSourceGenerator -> Generating package source for Humanizer.Core (v2.2.0)...
  Determining projects to restore...
  Restored C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj (in 5 ms).
  Restored C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGeneratorTask\PackageSourceGenera
  torTask.csproj (in 136 ms).
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018: The "G
etPackageItems" task failed unexpectedly.
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018: System
.ArgumentException: Found strong name key that doesn't map: Key=979442b78dfc278e, Assembly=Humanizer
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018:    at
Microsoft.DotNet.SourceBuild.Tasks.GetPackageItems.TryGetStrongNameData(AssemblyName assemblyName, StrongNameData& stro
ngNameData) in C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGeneratorTask\GetPackageItems
.cs:line 241
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018:    at
Microsoft.DotNet.SourceBuild.Tasks.GetPackageItems.SetCompileItems(PackageArchiveReader packageArchiveReader, TargetFra
meworkRegexFilter targetFrameworkRegexFilter) in C:\source-build-reference-packages\src\packageSourceGenerator\PackageS
ourceGeneratorTask\GetPackageItems.cs:line 153
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018:    at
Microsoft.DotNet.SourceBuild.Tasks.GetPackageItems.Execute() in C:\source-build-reference-packages\src\packageSourceGen
erator\PackageSourceGeneratorTask\GetPackageItems.cs:line 84
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018:    at
Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
C:\source-build-reference-packages\src\packageSourceGenerator\PackageSourceGenerator.proj(106,5): error MSB4018:    at
Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext ta
skLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
```

**Warnings:**
```
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'SQLitePCLRaw.core.dll' in any of the provided search d
irectories. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.csharp.work
spaces\3.11.0\microsoft.codeanalysis.csharp.workspaces.3.11.0.csproj::TargetFramework=netstandard2.0]
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'SQLitePCLRaw.batteries_v2.dll' in any of the provided
search directories. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.csh
arp.workspaces\3.11.0\microsoft.codeanalysis.csharp.workspaces.3.11.0.csproj::TargetFramework=netstandard2.0]
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'Humanizer.dll' in any of the provided search directori
es. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.csharp.workspaces\3
.11.0\microsoft.codeanalysis.csharp.workspaces.3.11.0.csproj::TargetFramework=netstandard2.0]
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'Humanizer.dll' in any of the provided search directori
es. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.csharp.workspaces\3
.11.0\microsoft.codeanalysis.csharp.workspaces.3.11.0.csproj::TargetFramework=netstandard2.0]
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'SQLitePCLRaw.core.dll' in any of the provided search d
irectories. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.workspaces.
common\3.11.0\microsoft.codeanalysis.workspaces.common.3.11.0.csproj::TargetFramework=netstandard2.0]
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'SQLitePCLRaw.batteries_v2.dll' in any of the provided
search directories. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.wor
kspaces.common\3.11.0\microsoft.codeanalysis.workspaces.common.3.11.0.csproj::TargetFramework=netstandard2.0]
C:\Users\mavasani\.nuget\packages\microsoft.dotnet.genapi.task\8.0.100-preview.6.23319.8\build\Microsoft.DotNet.GenAPI.
Task.targets(45,5): warning CP1002: Could not resolve reference 'Humanizer.dll' in any of the provided search directori
es. [C:\source-build-reference-packages\artifacts\bin\PackageSourceGenerator\microsoft.codeanalysis.workspaces.common\3
.11.0\microsoft.codeanalysis.workspaces.common.3.11.0.csproj::TargetFramework=netstandard2.0]
```